File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarSchemeWeekAddCommand.cs
using SIE.Resources.ShiftTypes;
using SIE.Web.Command;

namespace SIE.Web.Resources.CalendarSchemes.Commands
{
    /// <summary>
    /// 添加
    /// </summary>
    [JsCommand("SIE.Web.Resources.CalendarSchemes.Commands.CalendarSchemeWeekAddCommand")]
    public class CalendarSchemeWeekAddCommand : ViewCommand
    {
        /// <summary>
        /// 执行
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var shiftType = RT.Service.Resolve<ShiftTypeController>().GetDefaultShiftType();
            return shiftType;
        }
    }
}

File: Modules/Common/Resources/SIE.Web.Resources/CalendarSchemes/Commands/CalendarSchemeWeekAddCommand.js
SIE.defineCommand('SIE.Web.Resources.CalendarSchemes.Commands.CalendarSchemeWeekAddCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加", group: "edit" },

    onItemCreated: function (entity) {
        var model = entity.data;
        var me = this;
        this.view.execute({
            data: model,
            isSubmmit: false,
            success: function (res) {
                var data = res.Result;
                var year = new Date().getFullYear();
                var month = new Date().getMonth() + 1;
                var day = new Date().getDate() + 1;
                me.view.getCurrent().setActiveDate(year + "-" + month + "-" + day);
                if (data) {
                    me.view.getCurrent().setShiftTypeId_Display(data.Name);
                    me.view.getCurrent().setShiftTypeId(data.Id);
                }
            }
        }, me.view);
    }
});